lld/wasm/Symbols.h

```diff
@@ -495,26 +495,21 @@ class UndefinedTag : public TagSymbol {
 // and the lazy. We represent that with a lazy symbol with a weak binding. This
 // means that code looking for undefined symbols normally also has to take lazy
 // symbols into consideration.
-class LazySymbol : public Symbol {
+class LazyObject : public Symbol {
```

Comment:
I guess this is just copying the ELF backend, but why this renaming?

Reply:
Yes, the renaming is just to match ELF.. should a do that as it own separate PR to make this one more readable?

Reply:
Yeah, or maybe at least reference the corresponding ELF change if there was one single one.

Reply:
It seems like this was always just called LazyObject on the ELF side.

I'm not sure why `LazyObject` was used.. `LazySymbol` seems more descriptive.  I think maybe I will just revert this rename to keep this change small . We can always followup with a rename-only PR afterwards.

@MaskRay any ideas on this naming?  Does `LazyObject` make sense for some reason that I am missing?

Reply:
Oh.. I see what happened.  We used to have `LazyArchive` and `LazyObject` (two different kinds of lazy symbols) from http://reviews.llvm.org/D18814, but they were unified so now we are just left with `LazyObject`.  

I will instead propose we move ELF to `LazySymbol` which is more consistent with the other symbol types and more descriptive (IMHO).

Reply:
This PR is much smaller now.

Reply:
+1 on moving ELF to LazySymbol too, calling a symbol LazyObject was confusing to me.

Reply:
I think `LazySymbol` is better if `LazyArchive` is removed. Created #78809 to do the rename in the ELF port instead.